apps/vs-code-designer/src/app/utils/codeless/connection.ts

```diff
@@ -533,3 +561,385 @@ async function createAccessPolicyInConnection(
+    );
+    throw new Error('Missing WORKFLOWS_SUBSCRIPTION_ID or WORKFLOWS_RESOURCE_GROUP_NAME in local settings');
+  }
+
+  // Resolve connection ID with app settings
+  const resolvedConnectionId = connectionId.includes('@{appsetting(')
+    ? connectionId
+        .replace("@{appsetting('WORKFLOWS_SUBSCRIPTION_ID')}", subscriptionId)
+        .replace("@{appsetting('WORKFLOWS_RESOURCE_GROUP_NAME')}", resourceGroup)
+    : connectionId;
+
+  ext.outputChannel.appendLog(localize('resolvingConnection', 'Resolving connection ID: {0} -> {1}', connectionId, resolvedConnectionId));
+
+  // Create access policy
+  const objectIdSuffix = objectId.slice(-8);
+  const policyName = `${policyNamePrefix}-${objectIdSuffix}`;
+  const policiesUrl = `${formatSetting(baseManagementUri)}${resolvedConnectionId}/accessPolicies?api-version=2018-07-01-preview`;
+
+  // Check if policy already exists with improved logic
+  const policyExists = await checkExistingPolicy(policiesUrl, accessToken, objectId, tenantId, policyName, policyNamePrefix);
+
+  if (policyExists) {
+    return;
+  }
+
+  ext.outputChannel.appendLog(localize('policyNameGenerated', 'Generated policy name: {0} for user {1}', policyName, policyNamePrefix));
+  const apiVersion = '2016-06-01';
+  const addPolicyUrl = `${formatSetting(baseManagementUri)}${resolvedConnectionId}/accessPolicies/${encodeURIComponent(policyName)}?api-version=${apiVersion}`;
+
+  try {
+    await axios.put(
+      addPolicyUrl,
+      {
+        properties: {
+          principal: {
+            type: 'ActiveDirectory',
+            identity: { objectId, tenantId },
+          },
+        },
+      },
+      {
+        headers: {
+          Authorization: accessToken,
+          'Content-Type': 'application/json',
+        },
+      }
+    );
+
+    ext.outputChannel.appendLog(localize('policyCreated', 'Successfully created access policy for objectId: {0}', objectId));
+  } catch (error) {
+    const errorDetails = error.response?.data ? JSON.stringify(error.response.data) : error.message;
+    ext.outputChannel.appendLog(localize('policyCreationFailed', 'Failed to create access policy: {0}', errorDetails));
+  }
+}
+
+/**
+ * Checks if a policy exists for the given identity
+ * @returns true if policy exists and matches, false if it needs to be created
+ */
+async function checkExistingPolicy(
+  policiesUrl: string,
+  accessToken: string,
+  objectId: string,
+  tenantId: string,
+  policyName: string,
+  policyNamePrefix: string
+): Promise<boolean> {
+  try {
+    const response = await axios.get(policiesUrl, {
+      headers: { authorization: accessToken },
+      validateStatus: (status) => status < 500, // Don't throw for 4xx errors
+    });
+
+    // Handle different response statuses
+    if (response.status === 404) {
+      ext.outputChannel.appendLog(localize('noPoliciesFound', 'No access policies found for connection. Creating new policy.'));
```

Comment:
NIT: do we need to log those failing responses in telemetry?
- context.telemetry.error 
- ext.telemetryReporter.sendTelemetryEvent(eventName, { ...msg.data });